tests/graph/triangle/test_triangle_graph.py

```diff
@@ -0,0 +1,161 @@
+"""Tests for the TriangleGraph object."""
+
+import numpy as np
+import pytest
+from numpy.testing import assert_array_equal
+
+from landlab.graph.triangle.dual_triangle import DualTriGraph
+from landlab.graph.triangle.triangle_mesh import TriangleMesh
+
+if not TriangleMesh.validate_triangle():
+    pytestmark = pytest.mark.skip(reason="triangle is not installed")
+
+
+def test_graph_init():
+    graph = DualTriGraph(([0, 0, 10, 10], [0, 10, 10, 0]), triangle_opts="pqa1Djevz")
+
+    assert graph.number_of_nodes == 89
+    assert len(graph.x_of_node) == graph.number_of_nodes
+    assert len(graph.y_of_node) == graph.number_of_nodes
+    assert graph.number_of_links == 232
+    assert graph.number_of_patches == 144
+    assert graph.number_of_corners == graph.number_of_patches
+    assert graph.number_of_faces == 200
+    assert graph.number_of_cells == 57
+
+    assert np.all(graph.area_of_cell > 0.0)
+    assert np.all(graph.area_of_patch > 0.0)
+    assert np.all(graph.length_of_link > 0.0)
+    assert np.all(graph.length_of_face > 0.0)
```

Comment:
There are still some faces that have zero length. I believe this test is still failing.